### PR TITLE
Update service_vars with playbook set_stats data.

### PR DESCRIPTION
**Pull request by lfu**
https://bugzilla.redhat.com/show_bug.cgi?id=1734630

@miq-bot add_label changelog/yes, enhancement, ivanchuk/no, hammer/no

### CODE REVIEW

File: lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_ansible_method_base.rb [hunk 1]
```diff
@@ -80,6 +83,13 @@ def update_object_with_stats(key, value)
       object.object_send(:save!)
     end
 
+    def service_var_with_stats(key, value)
+      object = @workspace.current['service'] || @workspace.root['service']
+      raise _("Object update failed - service object not found for: [service_var__%{key} = %{value}]") % {:key => key, :value => value} if object.blank?
```
Comment (mkanoor) [inline lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_ansible_method_base.rb]: @lfu would it make sense to write a small helper function that did the normalization of variables names that adheres to the Ansible variable name restrictions.

**Comment by gmcculloug** [comment]
@mkanoor Please take another look.

**Comment by simaishi** [comment]
@lfu The BZ is now targeted for ivanchuk, can this be `ivanchuk/yes`?

**Comment by lfu** [comment]
This PR is based on https://github.com/ManageIQ/manageiq-automation_engine/pull/381 which should be back ported to `ivanchuk` too.

**Comment by simaishi** [comment]
Ivanchuk backport details:

```
$ git log -1
commit f678811278a43c4ee12d531eae4a68e6cc1fea71
Author: tina <tfitzger@redhat.com>
Date:   Tue Dec 17 10:27:05 2019 -0500

    Merge pull request #399 from lfu/service_vars_with_set_stats

    Update service_vars with playbook set_stats data.

    (cherry picked from commit 6bcb57c032f4fd6a581abf6c0932c687585846e8)

    https://bugzilla.redhat.com/show_bug.cgi?id=1734630
```

**Pull request merged**